src/sentry/tasks/integrations/migrate_alert_rules.py

```diff
@@ -0,0 +1,109 @@
+            if isinstance(p, OpsGeniePlugin) and p.is_configured(project):
+                plugin = p
+                break
+        if not plugin:
+            continue
+
+        # migrate key
+        api_key = plugin.get_option("api_key", project)
+        if api_key not in seen_keys:
+            seen_keys.add(api_key)
+            config = organization_integration.config
+            team_table = config["team_table"]
+            team = {
+                "team": project.name + " [MIGRATED]",
+                "id": str(organization_id) + "-" + project.name,
+                "integration_key": api_key,
+            }
+            team_table.append(team)
+            config.update({"team_table": team_table})
+            with transaction.atomic(router.db_for_write(OrganizationIntegration)):
+                integration_service.update_organization_integration(
+                    org_integration_id=organization_integration.id, config=config
+                )
+                logger.info(
+                    "api_key.migrated",
+                    extra={
+                        "integration_id": integration_id,
+                        "organization_id": organization_id,
+                        "project_id": project.id,
+                        "team_name": team["team"],
+                        "plugin": plugin.slug,
+                    },
+                )
+        else:
+            team = [team for team in team_table if team["integration_key"] == api_key][0]
+            logger.info(
+                "api_key.key_already_exists",
+                extra={
+                    "integration_id": integration_id,
+                    "organization_id": organization_id,
+                    "project_id": project.id,
+                    "team_name": team["team"],
+                    "plugin": plugin.slug,
+                },
+            )
+
+        # migrate alert rules
+        for rule in Rule.objects.filter(project_id=project.id):
+            actions = rule.data["actions"]
+            if ALERT_LEGACY_INTEGRATIONS in actions:
+                actions.append(
+                    {
+                        "id": "sentry.integrations.opsgenie.notify_action.OpsgenieNotifyTeamAction",
+                        "account": organization_integration.id,
+                        "team": team["id"],
+                    }
+                )
+                with transaction.atomic(router.db_for_write(Rule)):
+                    rule.save()
+                    logger.info(
+                        "alert_rule.migrated",
+                        extra={
+                            "integration_id": integration_id,
+                            "organization_id": organization_id,
+                            "project_id": project.id,
+                            "plugin": plugin.slug,
+                        },
+                    )
+
+        # disable plugin
+        plugin.disable(project)
```

Comment:
@leeandher this uninstalls vs disables—which one do we want?

Reply:
I think since we explicitly get user permission prior to this, It's fine to uninstall. It cleans up leftover user data from Sentry and it hides the 'installed' plugin from the UI (plus we've migrated all the data already)

Worth bringing up in stand up to get a second opinion though, if you feel disabling it may be better.

Reply:
Changed to uninstall.